pkg/controllers/nodepool/readiness/suite_test.go

```diff
@@ -112,4 +112,11 @@ var _ = Describe("Readiness", func() {
 		nodePool = ExpectExists(ctx, env.Client, nodePool)
 		Expect(nodePool.StatusConditions().IsTrue(status.ConditionReady)).To(BeFalse())
 	})
+	It("should have status condition on nodePool as not ready if nodeClass has a non zero deletion timestamp", func() {
+		nodeClass.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
```

Comment:
I don't think this works. You can't directly set the deletionTimestamp (the apiserver will reject this). You are going to have to add a dummy finalizer and then delete the object. There's other cases of doing this in other places in the Karpenter codebase